azure/durable_functions/models/DurableOrchestrationContext.py

```diff
@@ -158,19 +161,48 @@ def call_activity(self, name: str, input_: Optional[Any] = None) -> TaskBase:
         Task
             A Durable Task that completes when the called activity function completes or fails.
         """
+        if(isinstance(name, Callable) and not isinstance(name, FunctionBuilder)):
+            error_message = "The `call_activity` API received a `Callable` without an "\
+                "associated Azure Functions trigger-type. "\
+                "Please ensure you're using the Python programming model V2 "\
+                "and that your activity function is annotated with the `activity_trigger`"\
+                "decorator. Otherwise, provide in the name of the activity as a string."
+            raise ValueError(error_message)
+
+        if(isinstance(name, FunctionBuilder)):
+            try:
+                if (isinstance(name._function._trigger, ActivityTrigger)):
+                    name = name._function._name
+                else:
+                    error_message = "Received function with Trigger-type `"\
+                        + name._function._trigger.type\
+                        + "` but expected `ActivityTrigger`. Ensure your "\
+                        "function is annotated with the `activity_trigger` decorator "\
+                        "or directly pass in the name of the activity as a string."
+                    raise ValueError(error_message)
+            except AttributeError as e:
```

Comment:
Is this expected? It's not normal to check for internal consistency errors like this, but I don't have full context on the Python SDK. Also, I wonder if we should consider moving this try/catch logic in a common function to avoid duplication here and in `call_activity_with_retry`. It's a distraction from the core logic so moving it elsewhere would make the overall code more readable.

Reply:
+1 to refactoring some of this common logic onto a helper function.

Regarding why we have a try-catch: It's mostly because we're accessing "private" members from the Python worker library, as evidenced by the `_` prefixes. We do not expect those fields to be dropped or changed, and we'll be coordinating with the Python worker team to try to ensure they do not, but ultimately our integration with the Python V2 prog-model relies on accessing internals and, as such, it's a bit risky; thus the try-catch.